packages/framework-core/src/read-model-schema-migrator.ts

```diff
@@ -101,4 +126,60 @@ export class ReadModelSchemaMigrator {
   private static readModelSchemaVersion(readModel: ReadModelInterface): number {
     return readModel.boosterMetadata?.schemaVersion ?? 1
   }
+
+  private buildFilterForSearchReadModelsToMigrate(readModelName: string): FilterFor<ReadModelInterface> {
+    const expectedVersion = this.config.currentVersionFor(readModelName)
+    return {
+      or: [
+        {
+          boosterMetadata: {
+            schemaVersion: {
+              lt: expectedVersion,
+            },
+          },
+        },
+        {
+          boosterMetadata: {
+            schemaVersion: {
+              isDefined: false,
+            },
+          },
+        },
+      ],
+    }
+  }
+
+  private async searchReadModelsToMigrate(
+    readModelName: string,
+    filterFor: FilterFor<ReadModelInterface>,
+    limit: number,
+    cursor: undefined | Record<'id', string>
+  ): Promise<ReadModelListResult<ReadModelInterface>> {
+    return (await this.config.provider.readModels.search<ReadModelInterface>(
+      this.config,
+      readModelName,
+      filterFor,
+      {},
+      limit,
+      cursor,
+      true
+    )) as ReadModelListResult<ReadModelInterface>
+  }
+
+  private persistReadModel(newReadModel: ReadModelInterface, readModelName: string): Promise<unknown> {
+    const logger = getLogger(this.config, 'ReadModelSchemaMigrator#persistReadModel')
+    if (!(newReadModel && newReadModel.boosterMetadata)) {
+      throw new Error(`Error migrating ReadModel: ${newReadModel}`)
+    }
+    const currentReadModelVersion: number = newReadModel?.boosterMetadata?.version ?? 0
+    const schemaVersion: number =
+      newReadModel?.boosterMetadata?.schemaVersion ?? this.config.currentVersionFor(readModelName)
+    newReadModel.boosterMetadata = {
+      ...newReadModel?.boosterMetadata,
+      version: currentReadModelVersion + 1,
+      schemaVersion: schemaVersion,
+    }
+    logger.debug('Storing new version of read model', newReadModel)
+    return this.config.provider.readModels.store(this.config, readModelName, newReadModel, currentReadModelVersion)
```

Comment:
shouldn't you invoke `store` with the new read model version instead of `currentReadModelVersion` as the forth argument?